tests/std/tests/GH_004597_self_swap/test.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <utility>
+
+// Test GH-4597 "<utility>: Side effects in self-swaps of pair are skipped"
+struct swap_counter {
+    unsigned int* pcnt_ = nullptr;
+
+    friend constexpr void swap(swap_counter& lhs, swap_counter& rhs) noexcept {

```

Comment:
Note that the `constexpr` for `std::swap` is since C++20. Perphaps we should introduce this

```C++
#if _HAS_CXX20
#define CONSTEXPR20 constexpr
#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
#define CONSTEXPR20 inline
#endif // ^^^ !_HAS_CXX20 ^^^
```

and then apply `CONSTEXPR20` to the `swap` functions in this test.

Reply:
Done